controllers/configurationpolicy_controller.go

```diff
@@ -1045,277 +1216,255 @@ func (r *ConfigurationPolicyReconciler) handleDeletion(plc *policyv1.Configurati
+			message: fmt.Sprintf(
+				"The kind and apiVersion fields are required on the object template at index %d in policy %s",
+				index, plc.Name,
+			),
+		}
 
-		encryptionConfig, err = r.getEncryptionConfig(context.TODO(), plc)
-		if err != nil {
-			addTemplateErrorViolation("", err.Error())
+		return nil, nil, errEvent, nil
+	}
 
-			return parentStatusUpdateNeeded, err
+	scopedGVR, err := r.getMapping(objGVK, plc, index)
+	if err != nil {
+		errEvent := &objectTmplEvalEvent{
+			compliant: false,
+			reason:    "K8s error",
+			message:   err.Error(),
 		}
 
-		resolveOptions.EncryptionConfig = encryptionConfig
+		return nil, nil, errEvent, err
 	}
 
-	// set up raw data for template processing
-	var rawDataList [][]byte
+	relevantNamespaces := []string{}
 
-	isRawObjTemplate := false
+	// Fetch and filter namespaces using provided namespaceSelector
+	desiredNs := parsedMinMetadata.Metadata.Namespace
 
-	if plc.Spec.ObjectTemplatesRaw != "" {
-		rawDataList = [][]byte{[]byte(plc.Spec.ObjectTemplatesRaw)}
-		isRawObjTemplate = true
-	} else {
-		for _, objectT := range plc.Spec.ObjectTemplates {
-			rawDataList = append(rawDataList, objectT.ObjectDefinition.Raw)
-		}
+	// If the namespace is templated, consider it not explicitly set
+	if templates.HasTemplate([]byte(parsedMinMetadata.Metadata.Namespace), "", true) {
+		desiredNs = ""
 	}
 
-	resolveOptions.InputIsYAML = isRawObjTemplate
+	// The object is namespaced and either has no namespace specified or it is templated in the object definition
+	if scopedGVR.Namespaced && desiredNs == "" {
+		nsSelector := plc.Spec.NamespaceSelector
 
-	log.V(2).Info("Processing the object templates", "count", len(plc.Spec.ObjectTemplates))
+		selectedNamespaces, err := r.SelectorReconciler.Get(plc.Namespace, plc.Name, nsSelector)
+		if err != nil {
+			log.Error(err, "Failed to select the namespaces",
+				"namespaceSelector", nsSelector.String())
 
-	var tmplResolver *templates.TemplateResolver
+			msg := fmt.Sprintf("Error filtering namespaces with provided namespaceSelector: %v", err)
 
-	if usingWatch {
-		tmplResolver, err = templates.NewResolverWithDynamicWatcher(
-			r.DynamicWatcher, templates.Config{SkipBatchManagement: true},
-		)
-		objID := plc.ObjectIdentifier()
+			// only report this error if there wasn't another yet
+			errEvent := &objectTmplEvalEvent{
+				compliant: false,
+				reason:    "namespaceSelector error",
+				message:   msg,
+			}
 
-		resolveOptions.Watcher = &objID
-	} else {
-		tmplResolver, err = templates.NewResolverWithClients(
-			r.TargetK8sDynamicClient, r.TargetK8sClient.Discovery(), templates.Config{},
-		)
-	}
+			return nil, &scopedGVR, errEvent, err
+		}
 
-	if err != nil {
-		return parentStatusUpdateNeeded, err
+		if len(selectedNamespaces) != 0 {
+			relevantNamespaces = selectedNamespaces
+		}
+	} else if scopedGVR.Namespaced {
+		relevantNamespaces = []string{desiredNs}
+	} else {
+		// Cluster scoped
+		relevantNamespaces = []string{""}
```

Comment:
With this conditional, I think there's a risk here that a cluster-scoped objectDefinition that specifies a namespace would work? If that's correct, we probably shouldn't cover up that human error.

```suggestion
	} else {
		relevantNamespaces = []string{desiredNs}
	}
```

Reply:
That is correct, but we already ignore the namespace if it's provided for a cluster scoped resource in `getObject` and in the kubernetes-dependency-watches library.

So it'd be an additional change if we started returning an error here.

Reply:
Oh, didn't know that. Well alrighty then!